dotnet/src/webdriver/Interactions/ActionBuilder.cs

```diff
@@ -89,13 +91,26 @@ public override string ToString()
             return builder.ToString();
         }
 
+        private void ProcessTick(Interaction interaction)
+        {
+            ActionSequence sequence = this.GetOrAddSequence(interaction.SourceDevice);
+            sequence.AddAction(interaction);
+
+            foreach (KeyValuePair<InputDevice, ActionSequence> pair in this.sequences)
+            {
+                if (pair.Key != interaction.SourceDevice)
+                {
+                    pair.Value.AddAction(new PauseInteraction(pair.Key, TimeSpan.Zero));
+                }
+            }
+        }
```

Comment:
The `AddActions` method which takes an array of interactions is completely unused, so it's a lot of unnecessary processing to have `ProcessTicks(Interaction[])` called. This adds an overload to validate a single interaction, which is significantly simpler than the array version.

Reply:
outdated comment